src/lib/homestore_backend/heap_chunk_selector.cpp

```diff
@@ -131,6 +109,8 @@ csharedChunk HeapChunkSelector::select_specific_chunk(const pg_id_t pg_id, const
         auto& avalableBlkCounter = pg_it->second->available_blk_count;
         avalableBlkCounter.fetch_sub(vchunk.available_blks());
         remove_chunk_from_defrag_heap(vchunk.get_chunk_id());
+    } else {
```

Comment:
I am not sure when we will hit this else. maybe the check should be `if (vhunk)` ?

Reply:
it's depend on `exist` now